src/Command/Bulk/BulkCreateUsersAssignmentsCommand.php

```diff
@@ -42,63 +41,66 @@ public function __construct(
         parent::__construct(self::NAME, $ingesterSourceRegistry, $bulkCreateUsersAssignmentService);
     }
 
-    protected function configure()
+    protected function configure(): void
     {
         parent::configure();
 
         $this->setDescription('Responsible for creating user assignments based on user list (Local file, S3 bucket)');
     }
 
-    protected function process(
-        InputInterface $input,
-        ConsoleOutputInterface $consoleOutput,
-        int $batchSize,
-        bool $isDryRun
-    ): int {
-        $style = new SymfonyStyle($input, $consoleOutput);
-        $style->title('Simple Roster - Bulk Assignment Creation');
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        parent::initialize($input, $output);
 
-        $section = $consoleOutput->section();
-        $section->writeln('Starting assignment creation...');
+        $this->symfonyStyle->title('Simple Roster - Bulk Assignment Creation');
+    }
+
+    protected function process(OutputInterface $output): int
+    {
+        $this->symfonyStyle->note('Starting assignment creation...');
 
         try {
-            $source = $this->getIngesterSource($input);
             $bulkOperationCollection = new BulkOperationCollection();
 
+            $progressBar = $this->createNewFormattedProgressBar($output);
+
+            $progressBar->setMaxSteps($this->ingesterSource->count());
+            $progressBar->start();
+
             $numberOfProcessedAssignments = 0;
-            foreach ($source->getContent() as $row) {
+            foreach ($this->ingesterSource->getContent() as $row) {
                 $this->validateRow($row);
 
                 $operation = new BulkOperation($row['username'], BulkOperation::TYPE_CREATE);
 
-                $operation->setIsDryRun($isDryRun);
+                $operation->setIsDryRun($this->isDryRun);
 
                 $bulkOperationCollection->add($operation);
 
-                if (count($bulkOperationCollection) % $batchSize !== 0) {
+                if (count($bulkOperationCollection) % $this->batchSize !== 0) {
```

Comment:
`count($bulkOperationCollection)` is used miltiple times

Reply:
Fixed